npm-update.js

```diff
@@ -1,189 +1,88 @@
+var NpmStats   = require('npm-stats')
+var async      = require('async')
+var JSONStream = require('JSONStream')
 
 
-module.exports = function npm_update( options ){
+module.exports = function npmUpdate( options ){
   var seneca = this
 
-  options = seneca.util.deepextend({
-    task: null,
-    batchsize: 33,
+  var opts = Object.assign({
     registry: 'https://skimdb.npmjs.com/registry'
-  },options)
+  }, options)
 
+  //prevent multiple feed activations
+  var feedRunning = false
 
-  seneca.add( 'role:npm,task:registry_subscribe', registry_subscribe )
-  seneca.add( 'role:npm,task:process_modules',    process_modules )
-  seneca.add( 'role:npm,task:download_modules',   download_modules )
-
-
-  var state = {
-    registry_subscribe: { status:'stop' },
-    process_modules: { status:'stop' },
-    download_modules: { status:'stop' },
-  }
-  
-
-  function registry_subscribe( msg, respond ) {
-    var seneca = this
-
-    var feed
-
-    if( feed && 'stop' === msg.control ) {
-      state.registry_subscribe.status = 'stop'
-      feed.stop()
-    }
-
-    process_control( 
-      msg, respond, seneca, 'registry_subscribe', 
-      function( msg, respond, result ){
-        
-        state.registry_subscribe.status = 'run'
-        state.registry_subscribe.count = 0
-        state.registry_subscribe.since = new Date().toISOString()
-
-        feed = new follow.Feed({
-          db: options.registry,
-          since: 'now',
-        })
-
-        feed.on('error',function(error) {
-          result(error)
-        })
-
-        feed.on('change',function(change) {
-          state.registry_subscribe.count++
-          state.registry_subscribe.last = new Date().toISOString()
-          seneca.act('role:npm,info:change',{name:change.id})
-        })
-
-        feed.follow()
-      })  
-  }
-
-
-  function process_modules( msg, respond ) {
-    var seneca = this
-
-    process_control( 
-      msg, respond, seneca, 'process_modules', 
-      function( msg, respond, result ){
-        
-        state.process_modules.status = 'run'
-        state.process_modules.done = 0
-        state.process_modules.total = 0
-
-        seneca.make('mod').list$({done:false},function(err,list){
-          if(err) return result(err)
-          if( 'run' != state.process_modules.status ) return;
-
-          state.process_modules.total = list.length
-
-          async.eachLimit( list, options.batchsize, function(mod,done) {
-            if( 'run' != state.process_modules.status ) return;
-
-            var name = mod.id
-            seneca.act('role:npm,info:change',{name:name})
+  var feed = new follow.Feed({
+    db: opts.registry,
+    since: 'now'
+  })
 
-            seneca.make('mod',{id$:name}).load$(function(err,mod){
-              if( err ) return result(err)
+  feed.on('start', onFeedStart)
+  feed.on('stop', onFeedStop)
+  feed.on('error', (err) => {
+    console.log('feed error', err)
+    feed.stop() // try to restart???
```

Comment:
@AdrianRossouw Another place for fall off retries. Any idea on a module or recipe to handle?